### PR TITLE
Add openPMD-viewer

### PR DESCRIPTION
**Pull request by ax3l**
Add a recipe for openPMD-viewer.

Repo: https://github.com/openPMD/openPMD-viewer
Homepage: https://www.openPMD.org
License: BSD-3-Clause-LBNL

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by ax3l** [comment]
@RemiLehe I have listed the two of us as feedstock maintainer of this new package. Can you please confirm? :)

**Comment by RemiLehe** [comment]
Thanks! I confirm that I am happy to be maintainer for this feedstock.

File: recipes/openpmd-viewer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  url: https://github.com/openPMD/openPMD-viewer/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build:
+    - cython
```
Comment (xhochy) [inline recipes/openpmd-viewer/meta.yaml]: These should all go into build. But you probably need a compiler here?

File: recipes/openpmd-viewer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - numpy
+    - setuptools
+  host:
+    - python
+    - pip
+  run:
+    - h5py
+    - matplotlib
```
Comment (xhochy) [inline recipes/openpmd-viewer/meta.yaml]: Please use `matplotlib-base` to avoid hard dependencies on Qt.

File: recipes/openpmd-viewer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    # openPMD_notebook command
+    - jupyter
+    # examples
+    - wget  # [not win]
+
+test:
+  imports:
+    - opmd_viewer
+  commands:
+    - python -c "import opmd_viewer; print(opmd_viewer.__version__)"
```
Comment (xhochy) [inline recipes/openpmd-viewer/meta.yaml]: This is nearly the same as the import test above, can we omit this?

File: recipes/openpmd-viewer/meta.yaml (hunk 1 above)
Comment (ax3l) [reply recipes/openpmd-viewer/meta.yaml]: We currently pre-compile a `.pyx` file on install: https://github.com/openPMD/openPMD-viewer/blob/0.9.1/setup.py#L27-L31
Isn't this the right location for this?

File: recipes/openpmd-viewer/meta.yaml (hunk 2 above)
Comment (ax3l) [reply recipes/openpmd-viewer/meta.yaml]: Great idea, thanks!

File: recipes/openpmd-viewer/meta.yaml (hunk 3 above)
Comment (ax3l) [reply recipes/openpmd-viewer/meta.yaml]: That's true, no real added value here besides checking and prining the in-source version again (which I always like, too ^^).

File: recipes/openpmd-viewer/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/openpmd-viewer/meta.yaml]: You mean host I think

Comment (ax3l) [reply recipes/openpmd-viewer/meta.yaml]: So I will move only `setuptools` down and keep `cython` and `numpy` in `build`

Comment (xhochy) [reply recipes/openpmd-viewer/meta.yaml]: No, all dependencies listed here should go into `host`. `build` is for compilers & build-tools, `host` for build-time dependencies.

Comment (ax3l) [reply recipes/openpmd-viewer/meta.yaml]: Got ya, thanks a lot!

**Comment by ax3l** [comment]
@xhochy @scopatz thank you for reviewing. I updated the proposed feedstock with your suggestions.

**Pull request merged**